### PR TITLE
Add tests factory

### PR DESCRIPTION
**Pull request by AurelienFT**
Massa factory now has 73% coverage

### CODE REVIEW

File: massa-factory-worker/src/tests/tools.rs [hunk 1]
```diff
@@ -1,67 +1,60 @@
...
-use crate::start_factory;
+use crate::block_factory::BlockFactoryWorker;
+use crate::endorsement_factory::EndorsementFactoryWorker;
 use massa_wallet::test_exports::create_test_wallet;
 
 /// This structure store all information and links to creates tests for the factory.
-/// The factory will ask that to the the pool, consensus and factory and then will send the block to the consensus.
-/// You can use the method `new` to build all the mocks and make the connections
 #[allow(dead_code)]
-pub struct TestFactory {
```
Comment (Leo-Besancon) [inline massa-factory-worker/src/tests/tools.rs:27]: Is it not possible to remove the dead_code with a feature/cfg?

Comment (Leo-Besancon) [reply massa-factory-worker/src/tests/tools.rs:27]: (same for the endorsment test factory)

Comment (AurelienFT) [reply massa-factory-worker/src/tests/tools.rs:27]: Done

**Review by Leo-Besancon — DISMISSED** [review]

File: massa-factory-worker/src/tests/block_scenarios.rs [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        .times(1)
+        .return_once(move || parents);
+    consensus_controller
+        .expect_register_block()
+        .times(1)
+        .return_once(move |_, _, storage, created| {
+            assert!(created);
+            let block = storage.get_block_refs();
+            assert_eq!(block.len(), 1);
+            let &(ref lock, ref cvar) = &*pair2;
```
Comment (Eitu33) [inline massa-factory-worker/src/tests/block_scenarios.rs:53]: This is weird I'm pretty sure you don't need to do that as `let (ref lock, ref cvar) = *pair2;` would work the same. You've done it multiple time but LGTM besides that 👍

Comment (AurelienFT) [reply massa-factory-worker/src/tests/block_scenarios.rs]: Done I have updated following the example in the std : https://doc.rust-lang.org/stable/std/sync/struct.Condvar.html#examples-2 dunno why i had these refs.

**Review by Leo-Besancon — APPROVED** [review]

**Review by Eitu33 — APPROVED** [review]

**Comment by AurelienFT** [comment]
bors merge

**Pull request merged**